test/language_command_interface_tests.cpp

```diff
@@ -0,0 +1,161 @@
+	EXPECT_EQ(LanguageCommandInterface::UnitSystem::Metric, interfaceUnderTest.get_commanded_generic_units());
+
+	// This contains: "de", point, 12 hour time, ddmmyyyy, metric, no action, US, Metric, Reserved, Reserved, Imperial, Metric
+	std::uint8_t testData2[] = { 'd', 'e', 0b01011000, 0x00, 0b00111000, 0b10100100, 0xFF, 0xFF };
+
+	testMessage.set_data_size(0); // Resets the CAN message data vector
+	testMessage.set_data(testData2, 8);
+
+	interfaceUnderTest.process_rx_message(&testMessage, &interfaceUnderTest);
+	EXPECT_EQ("de", interfaceUnderTest.get_language_code());
+	EXPECT_EQ(LanguageCommandInterface::DecimalSymbols::Point, interfaceUnderTest.get_commanded_decimal_symbol());
+	EXPECT_EQ(LanguageCommandInterface::TimeFormats::TwelveHourAmPm, interfaceUnderTest.get_commanded_time_format());
+	EXPECT_EQ(LanguageCommandInterface::DateFormats::ddmmyyyy, interfaceUnderTest.get_commanded_date_format());
+	EXPECT_EQ(LanguageCommandInterface::DistanceUnits::Metric, interfaceUnderTest.get_commanded_distance_units());
+	EXPECT_EQ(LanguageCommandInterface::AreaUnits::NoAction, interfaceUnderTest.get_commanded_area_units());
+	EXPECT_EQ(LanguageCommandInterface::VolumeUnits::US, interfaceUnderTest.get_commanded_volume_units());
+	EXPECT_EQ(LanguageCommandInterface::MassUnits::Metric, interfaceUnderTest.get_commanded_mass_units());
+	EXPECT_EQ(LanguageCommandInterface::TemperatureUnits::Reserved, interfaceUnderTest.get_commanded_temperature_units());
+	EXPECT_EQ(LanguageCommandInterface::PressureUnits::Reserved, interfaceUnderTest.get_commanded_pressure_units());
+	EXPECT_EQ(LanguageCommandInterface::ForceUnits::ImperialUS, interfaceUnderTest.get_commanded_force_units());
+	EXPECT_EQ(LanguageCommandInterface::UnitSystem::Metric, interfaceUnderTest.get_commanded_generic_units());
+	EXPECT_LT(interfaceUnderTest.get_language_command_timestamp(), 100000);
+
+	testData2[6] = 75;
+	testData2[7] = 37;
+	testMessage.set_data_size(0); // Resets the CAN message data vector
+	testMessage.set_data(testData2, 8);
+	// Cover bad reserved bytes
+	interfaceUnderTest.process_rx_message(&testMessage, &interfaceUnderTest);
+
+	testData2[6] = 0xFF;
+	testData2[7] = 37;
+	testMessage.set_data_size(0); // Resets the CAN message data vector
+	testMessage.set_data(testData2, 8);
+	// Cover bad one bad reserved byte
+	interfaceUnderTest.process_rx_message(&testMessage, &interfaceUnderTest);
+
+	testData2[6] = 43;
+	testData2[7] = 0xFF;
+	testMessage.set_data_size(0); // Resets the CAN message data vector
+	testMessage.set_data(testData2, 8);
+	// Cover bad one bad reserved byte
+	interfaceUnderTest.process_rx_message(&testMessage, &interfaceUnderTest);
+
+	// Cover null message
+	interfaceUnderTest.process_rx_message(nullptr, &interfaceUnderTest);
+
+	// Cover null parent
+	interfaceUnderTest.process_rx_message(&testMessage, nullptr);
+
+	// Cover all null parameters
+	interfaceUnderTest.process_rx_message(nullptr, nullptr);
```

Comment:
For these 3, we can assert that values of language command interface remained unchanged